### PR TITLE
PID: Only apply temperature workaround to LUCCA M58 v2 models

### LINKED ISSUE #3: Remove workaround adding an extra degree to temperature control for LUCCA V3 hardware
## Context

In the past, the thermostat hardware used in [LUCCA M58 V2](https://clivecoffee.com/products/lucca-m58-espresso-machine-by-quick-mill) espresso systems would incorrectly report the user's entered temperature. If the system was using an LUCCA V2 thermostat, we'd subtract a degree from the reported temperature. In our firmware, we are classifying all LUCCA thermostats this way.

Now as of LUCCA M58 V3, they fixed that bug and we need to remove the workaround. We need to distinguish between the two thermostat versions and only apply the workaround to V2 hardware and not V3.

## Proposed Solution

In `PID.ts` in the method `applyRequestedTemperature` we need to check `DeviceContext.version` for v3 under the `LUCCA_M58` switch statement and only apply the temperature modification to v2.

### PR DESCRIPTION
**Pull request by kamranayub**
Closes #3

We were always applying a temperature change for any LUCCA M58 models but only v2 models actually need the workaround.

## Changes

- Modified switch statement in PID controller to only apply temperature change for V2 models

## TODOs

- [x] Add test coverage

### CODE REVIEW

File: src/PID.ts [hunk 1]
```diff
@@ -1,19 +1,21 @@
-import { DeviceContext } from './DeviceContext';
-import { DeviceMakeModel } from './Devices';
+import { DeviceContext } from "./DeviceContext";
```
[the remark this answers is not in the document]
Comment (kamranayub) [reply src/PID.ts]: OK, I'll update it shortly, thanks!

File: src/PID.ts [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
 
 export function applyRequestedTemperature(temp: number) {
-  const { makeAndModel } = DeviceContext;
+  const { makeAndModel, version } = DeviceContext;
 
   switch (makeAndModel) {
     case DeviceMakeModel.LUCCA_M58:
-      TemperatureContext.requestedTemperature = temp - 1;
-      break;
+      if (version === 2) {
```
Comment (kamranayub) [inline src/PID.ts]: TODO: Add a code comment explaining why this is being worked around

File: src/PID.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (kamranayub) [reply src/PID.ts]: I have reverted those formatting changes

[the remark this answers is not in the document]
Comment (wired-brain) [reply src/PID.ts]: Great, thanks!

**Comment by kamranayub** [comment]
@wired-cupofjoe OK, I've incorporated those v3 notes you added! 🤗

**Comment by wired-cupofjoe** [comment]
@kamranayub Fantastic! But I did just add one more change, I forgot a word. I did open a new PR #16 that contains the change but I accidentally *deleted my fork*! 😬 

Could you add that in too?

**Comment by kamranayub** [comment]
@wired-cupofjoe Done! Feeling pretty good about it ✨

**Comment by kamranayub** [comment]
Thanks @wired-cupofjoe. Hey @wired-brain I was just looking at #2 and thought I'd include those changes in here while I'm at it. Thanks!

**Comment by wired-brain** [comment]
@kamranayub We appreciate the initiative but we'd actually like to keep PRs tied to a single issue unless it makes sense to combine them. In this case, it touches two separate features so if you could split change out into a separate PR, that'd be great. Thanks!

**Comment by kamranayub** [comment]
@wired-brain Understood, I reverted the changes and split them off into #17

**Comment by wired-brain** [comment]
Excellent! I think the only issue left is fixing the tests. Can you check out the status and address any issues you find? Appreciate it!

**Pull request merged**